scripts/reset-simulators.rb

```diff
@@ -2,6 +2,9 @@
 
 require 'json'
 
+# use default DEVELOPER_DIR if not already set
```

Comment:
There are many changes in here that I assume came out of problems with the previous implementation. It'd be nice to know why each change was made. It's challenging to review this without that context.